src/config.js

```diff
@@ -276,9 +276,27 @@ export default class Config {
       networkConcurrency: this.networkConcurrency,
       networkTimeout: this.networkTimeout,
     });
-    this._cacheRootFolder = String(
-      opts.cacheFolder || this.getOption('cache-folder', true) || constants.MODULE_CACHE_DIRECTORY,
-    );
+
+    let cacheRootFolder = opts.cacheFolder || this.getOption('cache-folder', true);
```

Comment:
For the readers of this thread, if there are any: we have talked about getting a unique folder once we fallback to `tmp` like by using a timestamp or the PID of the process, or both, but then decided to do this: https://github.com/yarnpkg/yarn/pull/4143#discussion_r132509455

We should still probably try to ensure a unique folder in `tmp` when we fallback for safety.